src/locale/localed.c

```diff
@@ -438,10 +438,6 @@ static int method_set_vc_keyboard(sd_bus_message *m, void *userdata, sd_bus_erro
                         return sd_bus_error_setf(error, SD_BUS_ERROR_FAILED, "Keymap %s is not installed.", name);
         }
 
-        if (streq_ptr(keymap, c->vc_keymap) &&
-            streq_ptr(keymap_toggle, c->vc_keymap_toggle))
-                return sd_bus_reply_method_return(m, NULL);
```

Comment:
Instead of moving this down, try to convert the keymap earlier than the polkit call, then check if the coverted X11 keymap should be saved or not.